modules/devguide/examples/.clang-format

```diff
@@ -0,0 +1,11 @@
+# -*- mode: yaml; -*-
+
+BasedOnStyle: LLVM
+IndentWidth: 4
+BreakBeforeBraces: Linux
+ColumnLimit: 120
```

Comment:
Already asked about this in #116, but 

* seems like a very long line length. Realise we've inherited from existing code/settings, but worth mentioning in case we want to revisit.
* Why is this new file needed? There's already a .clang-format in the /c/ subdirectory?

Reply:
@osfameron The file I added here is the same as the one from `libcouchbase`:
https://github.com/couchbase/libcouchbase/blob/master/.clang-format

The tools we use (mainly CLion right now) can pick up this file for a project if you set the project settings. I'm going to remove the one in the `/c/` directory to avoid confusion, and also, because the higher level directory should be opened for the project (since it's CMake based now).